internal/namespaces/billing/v2alpha1/custom_billing_download.go

```diff
@@ -0,0 +1,235 @@
+package billing
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/scaleway/scaleway-cli/v2/internal/core"
+	"github.com/scaleway/scaleway-cli/v2/internal/interactive"
+	"github.com/scaleway/scaleway-cli/v2/internal/terminal"
+	billing "github.com/scaleway/scaleway-sdk-go/api/billing/v2alpha1"
+)
+
+const (
+	invoiceDefaultPrefix = "scaleway-invoice"
+)
+
+type billingDownloadRequest struct {
+	billing.DownloadInvoiceRequest
+	// extra arguments
+	FilePath     string
+	ForceReplace bool
+}
+
+func fileNameWithoutExtTrimSuffix(fileName string) string {
+	return strings.TrimSuffix(fileName, filepath.Ext(fileName))
+}
+
+func buildDownloadCommand(command *core.Command) *core.Command {
+	command.ArgsType = reflect.TypeOf(billingDownloadRequest{})
+	command.ArgSpecs = core.ArgSpecs{
+		{
+			Name:       "invoice-id",
+			Short:      `Invoice ID`,
+			Required:   true,
+			Deprecated: false,
+			Positional: false,
+		},
+		{
+			Name:       "file-type",
+			Short:      `Wanted file type`,
+			Required:   false,
+			Deprecated: false,
+			Positional: false,
+			EnumValues: []string{"pdf"},
+		},
+		{
+			Name:       "file-path",
+			Short:      `Wanted file locality`,
+			Required:   false,
+			Deprecated: false,
+			Positional: false,
+		},
+		{
+			Name:       "file-type",
+			Short:      `Wanted file extension`,
+			Required:   false,
+			Deprecated: false,
+			Positional: false,
+			Default: func(ctx context.Context) (value string, doc string) {
+				return billing.DownloadInvoiceRequestFileTypePdf.String(), "File extension `.pdf` as default"
+			},
```

Comment:
This is not how to use Default func

```suggestion
			Default: core.DefaultValueSetter(billing.DownloadInvoiceRequestFileTypePdf.String()),
```